File: common/src/main/java/munoon/bank/common/util/ValidationUtils.java
package munoon.bank.common.util;

import org.springframework.validation.FieldError;

import javax.validation.ConstraintViolation;
import java.util.*;

public class ValidationUtils {
    public static Map<String, List<String>> getErrorFieldMap(List<FieldError> fieldErrors) {
        Map<String, List<String>> result = new HashMap<>();
        fieldErrors.forEach(error -> {
            result.computeIfAbsent(error.getField(), k -> new ArrayList<>());
            result.get(error.getField()).add(error.getDefaultMessage());
        });
        return result;
    }

    public static Map<String, List<String>> getErrorFieldMap(Set<ConstraintViolation<?>> fieldErrors) {
        Map<String, List<String>> result = new HashMap<>();
        fieldErrors.forEach(error -> {
            String field = error.getPropertyPath().toString();
            result.computeIfAbsent(field, k -> new ArrayList<>());
            result.get(field).add(error.getMessage());
        });
        return result;
    }
}

File: user-resource-service/src/main/kotlin/munoon/bank/service/resource/user/util/ValidationUtils.kt
package munoon.bank.service.resource.user.util

import org.springframework.validation.FieldError
import javax.validation.ConstraintViolation

object ValidationUtils {
    fun getErrorFieldMap(fieldErrors: List<FieldError>): Map<String, List<String?>> {
        val map = HashMap<String, MutableList<String?>>()
        fieldErrors.forEach {
            map.computeIfAbsent(it.field) { ArrayList() }
            map[it.field]!!.add(it.defaultMessage)
        }
        return map
    }

    fun getErrorFieldMap(fieldErrors: Set<ConstraintViolation<*>>): Map<String, List<String?>> {
        val map = HashMap<String, MutableList<String?>>()
        fieldErrors.forEach {
            val field = it.propertyPath.toString()
            map.computeIfAbsent(field) { ArrayList() }
            map[field]!!.add(it.message)
        }
        return map
    }
}